pallets/services/src/lib.rs

```diff
@@ -600,12 +676,22 @@ pub mod module {
 				// remove the service request.
 				ServiceRequests::<T>::remove(request_id);
 
-				let service_id = NextInstanceId::<T>::get();
+				let service_id = Self::next_instance_id();
 				let operators = request
 					.operators_with_approval_state
 					.into_iter()
 					.map(|(v, _)| v)
 					.collect::<Vec<_>>();
+
+				// add the service id to the list of services for each operator's profile.
+				for operator in &operators {
+					OperatorsProfile::<T>::try_mutate_exists(&operator, |profile| {
+						profile
+							.as_mut()
+							.and_then(|p| p.services.try_insert(service_id).ok())
+							.ok_or(Error::<T>::NotRegistered)
```

Comment:
this makes every service have atleast N writes, N = number of operators, could be a problem on larger sets, can we reconsider the storage here, maybe represent operators inside the service

Reply:
Totally agree 💯 here, the approval process always has been the last one approves is the one creates it. We can move this to an `on_idle` hook or something like that, so it would be cheaper.

Reply:
Right now, there are no benchmarks as we're moving fast with the design, but I agree with you that this should be optimized later on.